### ISSUE TITLE
Override beam in dials.stills_process

### ISSUE THREAD

**Issue by biochem-fan** [dials/dials#231]
I have EIGER SRX datasets where information in the master HDF5 file is wrong. We can override the detector geometry in `dials.stills_process` by providing a reference JSON file. However, we cannot override the beam (wavelength and direction).

Thus, I would like to add an option `beam.wavelength` to the program. May I go ahead and send a pull request? Another, probably more comprehensive fix is to implement the same set of command-line arguments as `dials.import`, but I don't know how to do it without duplicating lots of codes.

**Comment by phyy-nx** [comment]
Hi, it should be possible to refactor things to use the same set of
parameters as dials.import.  But for now, sure, provide a pull request and
I'll review it.

**Comment by rjgildea** [comment]
This should be quite easy to add without too much code duplication - see
dials .modify_geometry for an example:

https://github.com/dials/dials/blob/master/command_line/modify_geometry.py

**Comment by biochem-fan** [comment]
Thank you very much for giving me the pointer. I made a fix. I will send PR tomorrow after tidying up.

I think geometry.*** arguments should have higher priority than the reference geometry JSON. Do you think this makes sense?

**Comment by biochem-fan** [comment]
I sent a pull request. I didn't know if I needed `set_XXX` as in modify_geometry.py.
`update_geometry(imageset)` updates `imageset` in place, doesn't it?

```
    imageset_new = update_geometry(imageset)
    imageset.set_detector(imageset_new.get_detector())
    imageset.set_beam(imageset_new.get_beam())
    imageset.set_goniometer(imageset_new.get_goniometer())
    imageset.set_scan(imageset_new.get_scan())
```

**Comment by phyy-nx** [comment]
Closed via #232.  Tested using the SALCA test case and changing wavelength with geometry.beam.wavelength=1.76846 (I.E., dropping units of precision), then grepping for wavelength in *.json and seeing them all be 1.76846.
